From 7d02b2509f5280bb7e65a65849e4e5c04c76a28c Mon Sep 17 00:00:00 2001
From: Gaming4JC <g4jc@hyperbola.info>
Date: Sun, 24 May 2020 00:27:45 -0400
Subject: [PATCH 07/12] init configure patch

---
 build/moz.configure/init.configure | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/build/moz.configure/init.configure b/build/moz.configure/init.configure
index 2e6034e8e..00d9935ba 100644
--- a/build/moz.configure/init.configure
+++ b/build/moz.configure/init.configure
@@ -656,9 +656,6 @@ def build_project(project, external_source_dir, build_env, help):
 
     build_app = project[0]
 
-    if not external_source_dir and build_app not in ('xulrunner', 'js'):
-        die('Cannot find project %s', build_app)
-
     build_app_abspath = os.path.join(base_dir, build_app)
 
     if exists(build_app_abspath):
-- 
2.25.2

