From c4f6159aa25117aa7c42c72bbbf69685257340d9 Mon Sep 17 00:00:00 2001
From: Gaming4JC <g4jc@hyperbola.info>
Date: Mon, 22 Jul 2019 20:49:53 -0400
Subject: [PATCH 1/1] Fix PGO Build

---
 Makefile.in            |  4 ++++
 gfx/thebes/gfxFont.cpp | 11 +++++++++++
 2 files changed, 15 insertions(+)

diff --git a/Makefile.in b/Makefile.in
index 4adf3ac9a..ed7581a04 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -397,4 +397,8 @@ config/export:
 
 endif
 
+
 # There used to be build interdependencies here. They are now in config/recurse.mk
+
+pgo-profile-run:
+	$(PYTHON) $(topsrcdir)/build/pgo/profileserver.py $(EXTRA_TEST_ARGS)
diff --git a/gfx/thebes/gfxFont.cpp b/gfx/thebes/gfxFont.cpp
index 490a866db..f2a61887b 100644
--- a/gfx/thebes/gfxFont.cpp
+++ b/gfx/thebes/gfxFont.cpp
@@ -2604,6 +2604,17 @@ gfxFont::GetShapedWord(DrawTarget *aDrawTarget,
     return sw;
 }
 
+template gfxShapedWord*
+gfxFont::GetShapedWord(DrawTarget *aDrawTarget,
+                       const uint8_t *aText,
+                       uint32_t    aLength,
+                       uint32_t    aHash,
+                       Script      aRunScript,
+                       bool        aVertical,
+                       int32_t     aAppUnitsPerDevUnit,
+                       uint32_t    aFlags,
+                       gfxTextPerfMetrics *aTextPerf);
+
 bool
 gfxFont::CacheHashEntry::KeyEquals(const KeyTypePointer aKey) const
 {
-- 
2.22.0

