diff --git a/build/moz.configure/init.configure b/build/moz.configure/init.configure
index 713b2ce61..fce447d36 100644
--- a/build/moz.configure/init.configure
+++ b/build/moz.configure/init.configure
@@ -631,8 +631,8 @@ def include_project_configure(project, external_source_dir, build_env, help):
 
     build_app = project[0]
 
-    if not external_source_dir and build_app not in ('xulrunner', 'js'):
-        die('Cannot find project %s', build_app)
+    #if not external_source_dir and build_app not in ('xulrunner', 'js'):
+    #    die('Cannot find project %s', build_app)
 
     path_project_src_dir_root = os.path.join(base_dir, build_app, 'moz.configure')

